lit_tests/kernel/wave/codegen.py

```diff
@@ -1934,3 +1935,113 @@ def get_item(a: tkl.Memory[M, N, ADDRESS_SPACE, tkl.f16]):
 
 
 # TODO: Add more tests once we have more than a stub implementation.
+
+
+@run_test
+def test_scalar_codegen_f32():
+    constraints: list[tkw.Constraint] = [
+        tkw.HardwareConstraint(
+            threads_per_wave=64,
+            waves_per_block=(1, 1, 1),
+            vector_shapes={M: 1, N: 27},
+        )
+    ]
+    constraints += [tkw.WorkgroupConstraint(M, 1, 1)]
+    constraints += [tkw.WorkgroupConstraint(N, 27, 0)]
+    constraints += [tkw.WaveConstraint(M, 1)]
+    constraints += [tkw.WaveConstraint(N, 27)]
+
+    @tkw.wave(constraints)
+    def scalar_codegen_f32(
+        a: tkl.Memory[M, N, ADDRESS_SPACE, tkl.f32],
+        c: tkl.f32,  # type: ignore
+        d: tkl.f32,  # type: ignore
+        b: tkl.Memory[M, N, ADDRESS_SPACE, tkl.f32],
+    ):
+        res = tkw.read(a)
+        c = tkw.broadcast(c, target_shape=[M, N])
+        d = tkw.broadcast(d, target_shape=[M, N])
+        temp = tkl.Register[M, N, tkl.f32](1.0) * c
+        res = res + temp + d
+        tkw.write(res, b)
+
+    options = WaveCompileOptions(
+        subs={
+            M: 1,
+            N: 27,
+            ADDRESS_SPACE: tkl.AddressSpace.GLOBAL_MEMORY.value,
+        },
+        canonicalize=True,
+        compile_to_mlir=True,
+    )
+    options = set_default_run_config(options)
```

Comment:
do we need default_run_config? seems like other lit test don't need it

Reply:
It is a mix. In some lit tests, it is used and in some it is not.